### DISCUSSION TITLE
Can I use Highs just to presolve and postsolve a MIP model?

### DISCUSSION THREAD

**Discussion by linpeng0105** [ERGO-Code/HiGHS#1926] [Q&A] [answered]
When I call highs in a C++ environment, can I use highs to presolve a MIP model, to get a reduced model?

Does Highs provide postsolve? to transform a reduced solution of the reduced model to an original solution of the original model?

**Comment by jajhall** [comment, marked as answer]
Yes, use `Highs::presolve()`,  `Highs::getPresolvedLp()`, `Highs::postsolve(const HighsSolution& solution)`, then `Highs::getSolution()`

**Comment by linpeng0105** [comment]
Thank you, I have another question:

A MIP problem A.mps, after presolve reduction generates a reduced problem B.mps.
I solved B.mps and obtained a solution B.sol, with an objective function value of Value(B).
After the postsolve process, B.sol was used to obtain A.sol, with an objective function value of Value(A).

My question: Is Value(A) equal to Value(B)?

**Reply by jajhall** [reply]
Yes, they are equal, but the objective function of the presolved problem will generally have a nonzero constant offset
